src/execution/mod.rs

```diff
@@ -431,6 +591,40 @@ where
             mems: memory_instances,
             globals: global_instances,
             data: data_sections,
+            tables,
+            elements,
+            passive_elem_indexes,
         }
     }
 }
+
+fn get_address_offset(value: Value) -> u32 {
+    match value {
+        Value::I32(val) => val,
+        Value::I64(val) => {
```

Comment:
Can you provide a source that attests that only I32s and I64 are generated by the initialiation expression of elements?

Reply:
Also, some source for the weird I64 behaviour would be nice

Reply:
I64 was wrong. Not even wasmtime implements it. They do that, though, for data section (as they implement the [Memory64 Proposal](https://github.com/WebAssembly/memory64)).